grammar/es/src/rules.rs

```diff
@@ -620,38 +975,141 @@ pub fn rules_time(b: &mut RuleSetBuilder<Dimension>) -> RustlingResult<()> {
-                      b.reg(r#"ma(?:ñ|n)ana"#)?,
+                      b.reg(r#"ma[ñn]ana"#)?,
                       |_| Ok(helpers::hour(4, false)?.span_to(&helpers::hour(12, false)?, false)?
                           .form(Form::PartOfDay(PartOfDayForm::Morning))
                           .latent())
     );
+    b.rule_1_terminal("beginning of morning",
+                      b.reg(r#"(?:pronto por|a primera hora (?:por|de)) la mañana|la mañana a primera hora"#)?,
+                      |_| Ok(helpers::hour(4, false)?
+                          .span_to(&helpers::hour(9, false)?, false)?
+                          .latent()
+                          .form(Form::PartOfDay(PartOfDayForm::Morning)))
+    );
+    b.rule_1_terminal("end of morning",
+                      b.reg(r#"[uúù]ltima hora de la mañana|la mañana a [uúù]ltima hora"#)?,
+                      |_| Ok(helpers::hour(10, false)?
+                          .span_to(&helpers::hour(12, false)?, false)?
+                          .latent()
+                          .form(Form::PartOfDay(PartOfDayForm::Morning)))
+    );
+    b.rule_1_terminal("lunch",
+                      b.reg(r#"(?:la )?(?:comida|hora de comer)"#)?,
+                      |_| Ok(helpers::hour(12, false)?
+                          .span_to(&helpers::hour(14, false)?, false)?
+                          .latent()
+                          .form(Form::Meal))
+    );
+    b.rule_1_terminal("after lunch",
+                      b.reg(r#"despu[eéè]s de (?:la hora de )?comer"#)?,
+                      |_| {
+                          let period = helpers::hour(13, false)?
+                              .span_to(&helpers::hour(17, false)?, false)?;
+                          Ok(helpers::cycle_nth(Grain::Day, 0)?.intersect(&period)?.form(Form::PartOfDay(PartOfDayForm::Afternoon)))
+                      }
+    );
+    b.rule_1_terminal("before lunch",
+                      b.reg(r#"antes de comer"#)?,
+                      |_| {
+                          let period = helpers::hour(10, false)?
+                              .span_to(&helpers::hour(12, false)?, false)?;
+                          Ok(helpers::cycle_nth(Grain::Day, 0)?.intersect(&period)?.form(Form::PartOfDay(PartOfDayForm::Morning)))
+                      }
+    );
+    b.rule_1_terminal("after work",
+                      b.reg(r#"después del trabajo|al salir de trabajar"#)?,
+                      |_| {
+                          let period = helpers::hour(17, false)?
+                              .span_to(&helpers::hour(21, false)?, false)?;
+                          Ok(helpers::cycle_nth(Grain::Day, 0)?.intersect(&period)?.form(Form::PartOfDay(PartOfDayForm::Evening)))
+                      }
+    );
     b.rule_1_terminal("afternoon",
-                      b.reg(r#"tarde"#)?,
+                      b.reg(r#"(?:la )?tarde"#)?,
                       |_| Ok(helpers::hour(12, false)?.span_to(&helpers::hour(19, false)?, false)?
```

Comment:
The resolutions of parts of days and meal times might have to be reviewed for Es (maybe for v2?).

Reply:
yes for sure